### PR TITLE
docs: Improve website + group articles + tweaks to documentation

### PR DESCRIPTION
**Pull request by olivroy**
Follow-up to #902 

* Improve function reference
* fix typos + links
* Correct cross-references based on the recent update
* Use [tidyverse style](https://style.tidyverse.org/documentation.html#r-code) guideline for references to functions. Enables linking on the website
* Move the raster to pkgdown/asset to allow users reproducing the example
* Move intro vignette to an article (there are many links + a widget which may cause problems on CRAN.
* Change headings to 1st and 2nd level when applicable.
* Reorder vignettes on the website
* Add leaflet.js copyright notice to the website
* Add `@keywords internal` for functions that are not necessarily user-facing to tidy up [reference index](https://rstudio.github.io/leaflet/reference/index.html)

TODO: eventually remove `vignettes/articles/nc` after successful build.

### CODE REVIEW

**Review by schloerke — COMMENTED** [review]
So sorry I didn't comment this earlier 🤦

File: R/normalize.R [hunk 1]
```diff
@@ -35,16 +35,22 @@ metaData.data.frame <- function(obj) obj
...
-#' @param lat latitude
+#' @param lng,lat longitude and latitude
 #' @param missingLng whether lng is missing
 #' @param missingLat whether lat is missing
 #' @param funcName Name of calling function (for logging)
+#' @returns
+#' * `derivePoints()` returns a lng/lat data frame.
+#' * `derivePolygons()` returns a list with bbox attributes.
 #' @export
+#' @keywords internal
```
[the remark this answers is not in the document]
Comment (schloerke) [reply R/normalize.R:53]: Answered in PR description.

Resolving

File: DESCRIPTION [hunk 2]
```diff
@@ -82,4 +82,3 @@ Config/Needs/website: dplyr, geojsonio, ncdf4, tidyverse/tidytemplate
 Encoding: UTF-8
 LazyData: true
 RoxygenNote: 7.3.1
-VignetteBuilder: knitr
```
[the remark this answers is not in the document]
Comment (schloerke) [reply DESCRIPTION]: Answered in description.

> Move intro vignette to an article (there are many links + a widget which may cause problems on CRAN

I'm all for less problems with CRAN

File: vignettes/articles/leaflet.Rmd [hunk 3]
```diff
@@ -1,10 +1,5 @@
 ---
 title: "Introduction to leaflet"
-output: rmarkdown::html_vignette
-vignette: >
-  %\VignetteIndexEntry{Introduction to leaflet}
-  %\VignetteEngine{knitr::rmarkdown}
-  %\VignetteEncoding{UTF-8}
```
[the remark this answers is not in the document]
Comment (schloerke) [reply vignettes/articles/leaflet.Rmd]: Answered in description.

> Move intro vignette to an article (there are many links + a widget which may cause problems on CRAN

I'm all for less problems with CRAN

File: R/plugin-awesomeMarkers.R [hunk 4]
```diff
@@ -159,13 +159,12 @@ makeAwesomeIcon <- function(
...
-#' Create a list of awesome icon data see
-#' \url{https://github.com/lennardv2/Leaflet.awesome-markers}
+#' Create a list of awesome icon data
 #'
 #' An icon can be represented as a list of the form \code{list(icon, library,
 #' ...)}. This function is vectorized over its arguments to create a list of
-#' icon data. Shorter argument values will be re-cycled. \code{NULL} values for
-#' these arguments will be ignored.
+#' icon data. Shorter argument values will be recycled. \code{NULL} values for
+#' these arguments will be ignored. See \url{https://github.com/lennardv2/Leaflet.awesome-markers}
```
Comment (schloerke) [inline R/plugin-awesomeMarkers.R:167]:
```suggestion
#' these arguments will be ignored. 
#' @seealso \url{https://github.com/lennardv2/Leaflet.awesome-markers}
```

File: vignettes/articles/widget.Rmd [hunk 5]
```diff
@@ -9,9 +9,9 @@ knitr::opts_chunk$set(
 )
 ```
 
-The function `leaflet()` returns a Leaflet map widget, which stores a list of objects that can be modified or updated later. Most functions in this package have an argument `map` as their first argument, which makes it easy to use the pipe operator `%>%` in the **magrittr** package, as you have seen from the example in the [Introduction](./).
+The function `leaflet()` returns a Leaflet map widget, which stores a list of objects that can be modified or updated later. Most functions in this package have an argument `map` as their first argument, which makes it easy to use the pipe operator `%>%` in the **magrittr** package, as you have seen from the example in `vignette("leaflet")`.
```
Comment (schloerke) [inline vignettes/articles/widget.Rmd:12]: Still keep the link to the article...

```suggestion
The function `leaflet()` returns a Leaflet map widget, which stores a list of objects that can be modified or updated later. Most functions in this package have an argument `map` as their first argument, which makes it easy to use the pipe operator `%>%` in the **magrittr** package, as you have seen from the example in [`vignette("leaflet")`](./).
```

File: vignettes/articles/choropleths.Rmd [hunk 6]
```diff
@@ -63,6 +63,7 @@ We'll start by loading the data from JSON. While the Leaflet.js example loads th
 
 In this case, we'll use the `geojsonio` package to load the data into `sp` objects, which will let us easily manipulate the geographic features, and their properties, in R.
 
+<!-- TODO Use sf everywhere? -->
```
[the remark this answers is not in the document]
Comment (schloerke) [reply vignettes/articles/choropleths.Rmd:66]: Let's save this for a followup PR?

**Comment by schloerke** [comment]
> TODO: eventually remove vignettes/articles/nc after successful build.

If this is just for gh-pages, I'm happy to make an issue for a followup PR later. `gh-pages` isn't in the main branch, so it can have many copies of files

File: vignettes/articles/widget.Rmd (hunk 5 above)
Comment (olivroy) [reply vignettes/articles/widget.Rmd:12]: `vignette()` autolinks on pkgdown ;)  https://pkgdown.r-lib.org/articles/pkgdown.html#metadata See the link there to `vignette("customize")`  
image

**Comment by schloerke** [comment]
Sorry @olivroy , you'll have to run `devtools::document()` to get rid of the failing docs check.

File: vignettes/articles/widget.Rmd (hunk 5 above)
Comment (schloerke) [reply vignettes/articles/widget.Rmd:12]: Thank you for another TIL!

**Resolving without accepting suggestion

File: vignettes/articles/choropleths.Rmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (olivroy) [reply vignettes/articles/choropleths.Rmd:66]: With the archival of legacy spatial packages, sf and terra are the new "best practices" packages. sp is still there, but uses sf under the hood  https://r-spatial.org/r/2023/04/10/evolution3.html

**Review by schloerke — APPROVED** [review]

**Comment by schloerke** [comment]
⭐ Thank you @olivroy !! ⭐

**Pull request merged**